lib/compliance/blackduck.js

```diff
@@ -0,0 +1,244 @@
+    if ('\n' in text) {
+      let re = /\\/gi;
+      text = text.replace(re, '');
+      let lines = text.split('\n').splice(1) // pull off copyright text
+      let license = ' '.join(lines)
+      if (license.indexOf(BSD) !== -1) {
+        return 'BSD'
+      }
+    }
+   */
+    return 'UNKNOWN'
+  },
+
+  /**
+   * We have a little chunk of the license data that we have gotten from nlf
+   * now we want to do our best to tag it
+   * @param {Object} licenseSources - the section of the nlf config with license data
+   * @returns {String} - some pithy string summary of what the licenses are (e.g. "MIT, BSD")
+   */
+  _getLicense (licenseSources) {
+    let output = []
+    if (licenseSources.package.sources.length > 0) {
+      for (let source of licenseSources.package.sources) {
+        output.push(source.license)
+      }
+    } else if (licenseSources.license.sources.length > 0) {
+      for (let source of licenseSources.license.sources) {
+        output.push(this._parseLicense(source))
+      }
+    }
+    return output.join(', ')
+  },
+
+  /**
+   * Get all the version info in a specific format
+   * @param {String} cwd - current working directory
+   * @param {String} npmOutputFilePath - path where we're going to write the package dependencies file
+   * @returns {Promise} A promise for a string
+   */
+  getNpmLicenseData (cwd, npmOutputFilePath) {
+    console.log('...getting license data')
+    return this.findLicenses({
+      directory: cwd,
+      production: true
+    })
+      .then((packageLicenseData) => {
+        let versionData = []
+        for (let packageLicenseDatum of packageLicenseData) {
+          if (packageLicenseDatum.licenseSources === undefined) {
+            logger.log(`ERROR: ${packageLicenseDatum.id} has no licenseSources?`)
+            continue
+          }
+          let license = this._getLicense(packageLicenseDatum.licenseSources)
+          versionData.push({version: packageLicenseDatum.version, name: packageLicenseDatum.name, license})
+        }
+        versionData = _.sortBy(versionData, ['name', 'version'])
+
+        return this.writeFile(npmOutputFilePath, JSON.stringify({'schema-version': 1, data: versionData}, null, '  '))
+          .then(() => {
+            return `successfully wrote ${npmOutputFilePath}`
+          })
+          .catch((error) => {
+            logger.log(`(1) ERROR: writing ${npmOutputFilePath}`, error)
+            throw error
+          })
+      })
+  },
+
+  /**
+   * Get a list of repositories that don't contain code we care about
+   * @param {String} cwd - string for current-working-directory
+   * @param {String} ourRepoName - the name of our own repos
+   * @returns {Promise} A promise for a string of repository URLs
+   */
+  getPackageData (cwd, ourRepoName) {
+    console.log('...getting package data')
+    const publicRegex = /\s+"(ember\-frost\-\S+)"/
+    const privateRegex = /\s+"(frost\-\S+)"/
```

Comment:
This stuff will need to be moved to options within `.pr-bumper.json`, we shouldn't hard-code our own regex for what is public w/i the company and private w/i the company.